### ISSUE TITLE
Adam optimizer countered by learning rate reset each iteration

### ISSUE THREAD

**Issue by johnjaniczek** [NVIDIA/tacotron2#445]
Thanks for this great implementation of Tacotron2. I just have one question/issue I want to bring up.

To the best of my knowledge the Adam optimizer is not actually scheduling the learning rate because lines 208 to 211 of train.py imply that the learning rate is getting reset each iteration:
```
        for i, batch in enumerate(train_loader):
            start = time.perf_counter()
            for param_group in optimizer.param_groups:
                param_group['lr'] = learning_rate
```

Now, the model trains fine - so it's not really a bug. But it does cause me a bit of confusion as the optimizer is instantiated as Adam which gives the expectation that the learning rate will adapt each iteration. It seems like instead we are using SGD, which is fine - but it's a bit confusing. If this is true, I would recommend either just instantiating an SGD optimizer or removing the lines of code which reset the learning rate each iteration. 

Sorry if I am incorrect on how the optimizer is working. I'll admit that this may just be the case of me misinterpreting something. Anyway, hope to hear back from you. Cheers!

**Comment by johnjaniczek** [comment]
Okay, I read into torch.optim.Adam a bit more and it appears that the 'lr' parameter is constant even though the step sizes are adaptive (weighting steps by both lr and the other adaptive parameters). So, I was wrong in that this should still allow Adam optimization. Although there is still no reason to reset `lr` since it is constant, it shouldn't affect training in any way.
